### PR TITLE
Minimum Group Owners Required for a Project

### LINKED ISSUE #3: Minimal number of owners required for a vault
It might be useful to specify, as part of the data retention policy, the minimal number of owners (i.e., those who will get actionable e-mails) that a group must have before a vault can be created.

### PR DESCRIPTION
**Pull request by michael-grace**
Closes #3 

This introduces a configuration option `min_group_owners` that is the minimum number of owners defined in the identity manager (i.e. LDAP) the project group must have before vault/sandman can be used on the project

### CODE REVIEW

**Comment by sb10** [comment]
[Ensure a group has 3 owners before sandman runs on it](https://trello.com/c/6pcyIaA3/61-ensure-a-group-has-3-owners-before-sandman-runs-on-it)

File: test/api/vault/test_vault.py [hunk 1]
```diff
@@ -551,5 +555,31 @@ def test_incorrect_parent_directory_permissions(self):
...
+
+    def test_create_vault_not_enough_group_owners(self):
+        self.assertRaises(
+            VaultExc.MinimumNumberOfOwnersNotMet,
+            Vault, relative_to=self._path, idm=_DummyIdM(
+                1, num_owners=int(config.min_group_owners) - 1
+            )
+        )
+
+    def test_create_vault_enough_group_owners(self):
```
Comment (sb10) [inline test/api/vault/test_vault.py:531]: Add a test that proves we can have more the minimum and it's still OK.

File: test/bin/sandman/test_walk.py [hunk 2]
```diff
@@ -45,18 +48,19 @@ class _DummyGroup(IdM.base.Group):
...
+    def __init__(self, gid, owner, member=None, num_owners: int = 3):
         self._id = gid
         self._owner = owner
         self._member = member or owner
+        self._num_owners = num_owners
 
     @property
     def name(self):
-        raise NotImplementedError
+        return f"dummy group {self._id}"
```
Comment (sb10) [inline test/bin/sandman/test_walk.py]: Is it possible to refactor the dummy stuff used in the different test files so they all use the same implementation, rather than re-implement these changes multiple times?

**Review by sb10 — CHANGES_REQUESTED** [review]
Would be good to refactor the dummy bits so all tests used the same implementation.

**Review by sb10 — DISMISSED** [review]

**Review by sb10 — APPROVED** [review]
